econf/econf.py

```diff
@@ -1,8 +1,22 @@
 import ctypes.util
 from enum import Enum
+from dataclasses import dataclass
 from typing import *
 from ctypes import *
 
+LIBNAME = ctypes.util.find_library("econf")
+LIBECONF = CDLL(LIBNAME)
+
+@dataclass
+class Keyfile:
+    """
+    Class which points to the Key Value storage object
+    """
+    ptr: c_void_p
```

Comment:
(sorry, I forgot to mention this early)

This `ptr` should not be used directly by the user, I guess.  I suggest renaming it to `__ptr`, so it is clear that should be managed as private

I like the idea of returning an object that will be used only for the destructor (even if this is not guaranteed by Python)

Do you plan to add more methods there?

The name `Keyfile` can also a bit confusing. Is not representing a key (but a configuration), and is not a file (but a handle or opaque pointer).  In econf seems to be the `struct econf_file`, maybe we can use this name: `EconfFile`

Reply:
ok, no problem, keyfile is ok too